components/datadog/fakeintake/docker.go

```diff
@@ -0,0 +1,55 @@
+package fakeintake
+
+import (
+	"log"
+	"net"
+
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/DataDog/test-infra-definitions/components"
+	"github.com/pulumi/pulumi-docker/sdk/v4/go/docker"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+const hostPort = 30080
+
+func NewLocalDockerFakeintake(e config.CommonEnvironment, resourceName string) (*Fakeintake, error) {
+	return components.NewComponent(e, resourceName, func(comp *Fakeintake) error {
+
+		_, err := docker.NewContainer(e.Ctx, e.CommonNamer.ResourceName("local-docker-container"), &docker.ContainerArgs{
+			Image: pulumi.String("public.ecr.aws/datadog/fakeintake:latest"),
+			Ports: docker.ContainerPortArray{
+				&docker.ContainerPortArgs{
+					Internal: pulumi.Int(80),
+					External: pulumi.Int(hostPort),
+				},
+			},
+		}, e.WithProviders(config.ProviderDocker))
+		if err != nil {
+			return err
+		}
+
+		localIP := getLocalIP()
+
+		comp.Host = pulumi.Sprintf(localIP.String())
+		comp.Port = hostPort
+		comp.Scheme = "http"
+		comp.URL = pulumi.Sprintf("%s://%s:%d", comp.Scheme, comp.Host, comp.Port)
+
+		return nil
+	})
+}
+
+func getLocalIP() net.IP {
+	// Open a connection to an external valid URL to
+	// get the local address from the connection instance
+	// The URL does not need to exist
+	conn, err := net.Dial("udp", "8.8.8.8:80")
+	if err != nil {
+		log.Fatal(err)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Should you rather return the error, instead of panicking?